ci/scripts/c_glib_build.sh

```diff
@@ -21,7 +21,10 @@ set -ex
 
 source_dir=${1}/c_glib
 build_dir=${2}/c_glib
-: ${ARROW_GLIB_GTK_DOC:=false}
+build_root=${2}
+with_docs=${3:-false}
+
+: ${ARROW_GLIB_GTK_DOC:=${with_docs}}
```

Comment:
We can remove `ARROW_GLIB_GTK_DOC` and specify `-Dgtk_doc=${with_docs}` for `meson`.